virtualmachine/vsphere/util.go

```diff
@@ -833,19 +832,7 @@ var cloneFromTemplate = func(vm *VM, dcMo *mo.Datacenter, usableDatastores []str
 		config.DeviceChange = append(config.DeviceChange, conf...)
 	}
 
-	err = checkAndCreateCustomSpec(vm)
```

Comment:
If we are not using checkAndCreateCustomSpec() anymore, then we should remove it along with other functions which were only called in checkAndCreateCustomSpec() such as createCustomSpecStaticIp.